govcd/defined_interface.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2022 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"net/url"
+	"strings"
+)
+
+// DefinedInterface is a type for handling Defined Interfaces, from the Runtime Defined Entities framework, in VCD.
+type DefinedInterface struct {
+	DefinedInterface *types.DefinedInterface
+	client           *Client
+}
+
+// CreateDefinedInterface creates a Defined Interface.
+// Only System administrator can create Defined Interfaces.
+func (vcdClient *VCDClient) CreateDefinedInterface(definedInterface *types.DefinedInterface) (*DefinedInterface, error) {
```

Comment:
I am curious, how will other RDE types be called (by definition of API)? Is there a common prefix to them?

CC: @Didainius @dataclouder @adezxc

Reply:
The others are `DefinedEntityType` and `DefinedEntity`.
In this case I followed the same naming used here:

- https://developer.vmware.com/apis/vmware-cloud-director/v37.1/defined-interface/
- https://developer.vmware.com/apis/vmware-cloud-director/v37.1/defined-entity-type/
- https://developer.vmware.com/apis/vmware-cloud-director/v37.1/defined-entity/

But I'm open for a renaming if it sounds clearer/aids grouping functionality.

Reply:
Thanks, being inline with API is good!